### PR TITLE
feat: Indicate user with valid E2EI certificate (WPB-3228)

### PR DESCRIPTION
**Pull request by borichellow**
# What's new in this PR?

### Issues

No way to get users E2EI certificate status (only for each client separately)

### Causes (Optional)

Was not implemented

### Solutions

Added 2 useCases: 
1. For getting users E2EI certificate status by `UserId` ( `GetUserE2eiCertificateStatusUseCase` )
2. For getting users E2EI certificate status for all the users in conversation, by `ConversationId` ( `GetMembersE2EICertificateStatusesUseCase` )

And Tests for these UseCases

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt [hunk 1]
```diff
@@ -551,6 +555,42 @@ internal class MLSConversationDataSource(
             }
         }
 
+    override suspend fun getUserIdentity(userId: UserId) =
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt:558]: Suggestion: we use this function almost every time we want to show the badge in a conversation's members list or in a 1:1 conversation, so can we get the conversation id from the AR when we calling this function?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt [hunk 2]
```diff
@@ -551,6 +555,42 @@ internal class MLSConversationDataSource(
...
+                wrapMLSRequest {
+                    mlsClient.getUserIdentities(
+                        mlsGroupId,
+                        listOf(userId.toCrypto())
+                    )[userId.value]!!
+                }
+            }
+        }
+
+    override suspend fun getMembersIdentities(conversationId: ConversationId): Either<CoreFailure, Map<UserId, List<WireIdentity>>> =
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt:570]: question: when we fetch the users, are they in the same order we show in the UI for the conversation's member?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt (hunk 1 above)
Comment (borichellow) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt:558]: actually that function is used only in `OtherUserProfileScreen`. And there is no `conversationId` for the direct conversation with that user from the beginning.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt (hunk 2 above)
Comment (borichellow) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt:570]: I'm not user about order in List, that why used `Map<UserId, List<WireIdentity>>` here :)

File: persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq [hunk 3]
```diff
@@ -380,6 +380,17 @@ ON Member.conversation = Conversation.qualified_id OR Client.user_id = Conversat
 WHERE Conversation.mls_group_id IS NOT NULL AND Client.id = :clientId
 ORDER BY Conversation.type DESC LIMIT 1;
 
+getMLSGroupIdByUserId:
+SELECT Conversation.mls_group_id FROM Member
+JOIN Conversation ON Conversation.qualified_id = Member.conversation
+WHERE Conversation.type = 'ONE_ON_ONE' AND Member.user = :userId;
```
Comment (mchenani) [inline persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq:386]: quesion: what if we don't have a 1:1 with that user? and only some mls groups in common?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt [hunk 4]
```diff
@@ -551,6 +558,41 @@ internal class MLSConversationDataSource(
...
+                wrapMLSRequest {
+                    mlsClient.getUserIdentities(
+                        mlsGroupId,
+                        listOf(userId.toCrypto())
+                    )[userId.value]!!
+                }
+            }
+        }
+
+    override suspend fun getMembersIdentities(
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt:573]: question: from the place your passing the conversationId [from the AR] you can make sure that you only pass the groupId, and if the protocol is not MLS you can't call it with a normal conversation Id, wdyt?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt [hunk 5]
```diff
@@ -551,6 +558,41 @@ internal class MLSConversationDataSource(
...
+        conversationId: ConversationId,
+        userIds: List<UserId>
+    ): Either<CoreFailure, Map<UserId, List<WireIdentity>>> =
+        wrapStorageRequest {
+            conversationDAO.getMLSGroupIdByConversationId(conversationId.toDao())!!
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt:578]: then we don't need to call this function!

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt [hunk 6]
```diff
@@ -551,6 +558,41 @@ internal class MLSConversationDataSource(
...
+        wrapStorageRequest { conversationDAO.getMLSGroupIdByUserId(userId.toDao()) }.flatMap { mlsGroupId ->
+            mlsClientProvider.getMLSClient().flatMap { mlsClient ->
+                wrapMLSRequest {
+                    mlsClient.getUserIdentities(
+                        mlsGroupId,
+                        listOf(userId.toCrypto())
+                    )[userId.value]!!
+                }
+            }
+        }
```
Comment (ohassine) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt:571]: Can we cover it with unit test ?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt [hunk 7]
```diff
@@ -551,6 +558,41 @@ internal class MLSConversationDataSource(
...
+                        .forEach { (userIdValue, identities) ->
+                            userIds.firstOrNull { it.value == userIdValue }?.also {
+                                userIdsAndIdentity[it] = identities
+                            }
+                        }
+
+                    userIdsAndIdentity
+                }
+            }
+        }
```
Comment (ohassine) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt:594]: Also this one, unit test is missing

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/e2ei/GetMembersE2EICertificateStatusesUseCaseTest.kt [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+import io.mockative.any
+import io.mockative.eq
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class GetMembersE2EICertificateStatusesUseCaseTest {
+
+    @Test
+    fun givenErrorOnGettingMembersIdentities_thenEmptyMapResult() = runTest {
```
Comment (ohassine) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/e2ei/GetMembersE2EICertificateStatusesUseCaseTest.kt:40]: For consistency, can we follow the pattern `given, when, then` ?

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/e2ei/GetUserE2eiCertificateStatusUseCaseTest.kt [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+import io.mockative.eq
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class GetUserE2eiCertificateStatusUseCaseTest {
+
+    @Test
+    fun givenErrorOnGettingUserIdentity_thenNotActivatedResult() = runTest {
```
Comment (ohassine) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/e2ei/GetUserE2eiCertificateStatusUseCaseTest.kt:41]: Also for here

**Pull request merged**